subprojects/base-services/src/integTest/groovy/org/gradle/internal/exceptions/DefaultMultiCauseExceptionIntegrationTest.groovy

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.exceptions
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class DefaultMultiCauseExceptionIntegrationTest  extends AbstractIntegrationSpec {
+    def 'when tasks throw exceptions that offer resolutions, those resolutions are included'() {
+        given:
+        buildFile << """
+            ${defineTestException()}
+
+            tasks.register('myTask') {
+                doLast {
+                    throw new TestResolutionProviderException('resolution1')
+                }
+            }
+        """
+
+        when:
+        fails 'myTask'
+
+        then:
+        assertHasSuggestions('resolution1')
+    }
+
+    def 'when tasks throw multi cause exceptions with resolutions offered by the causes, those resolutions are included'() {
+        given:
+        buildFile << """
+            ${defineTestException()}
+
+            tasks.register('myTask') {
+                doLast {
+                    def fail1 = new TestResolutionProviderException('resolution1')
+                    def fail2 = new TestResolutionProviderException('resolution2')
+                    throw new org.gradle.internal.exceptions.DefaultMultiCauseException('failure', fail1, fail2)
```

Comment:
What happens if the same resolution is reported twice?

Reply:
Only put on the command line once.  I think this is fine, why report duplicates?  I'll add a test to demonstrate this.